src/components/feedback/feedback.js

```diff
@@ -13,85 +14,129 @@ const environment =
+    // sets event to the state and marks feedbackSent as true
+    this.setState(
+      { feedbackSent: true, event: this.createSegmentEvent() },
+      () => {
+        // sends event to Segment
+        this.sendToSegment();
+        // if enabled, sends feedback to Sentry
+        if (this.props.feedbackSentryDsn !== false) {
+          this.sendToSentry();
+        }
+      }
+    );
+  }
+
+  // create event object to sent to Segment
+  createSegmentEvent() {
+    return {
       event: 'Sent docs feedback',
+      // set user if available (needed for forward-event request)
+      userId: this.props.userName || undefined,
+      ...(!this.props.userName && { anonymousId: anonymousId }),
       properties: {
-        helpful: this.state.helpful, // true, false
-        site: this.props.site, // name of current site, helpful for filtering in Mode
-        section: this.props.section || undefined, // (optional) name of section for longer pagers, helpful for fitering in Mode and identifying section areas
-        feedback: this.state.feedback, // (optional) textarea feedback
-        page: this.props.location || undefined, // get page context
-        userId: this.props.userName || undefined, // set user if available
-        environment, // staging or production
-        location // pull full window.location
+        // true, false,
+        helpful: this.state.helpful,
+        // text feedback
+        ...(this.state.feedback && { feedback: this.state.feedback }),
+        // name of current site, helpful for filtering in Mode
+        site: this.props.site,
+        // (optional) name of section for longer pagers, helpful for fitering in Mode and identifying section areas
+        section: this.props.section || undefined,
+        // get page context
+        page: this.props.location || undefined,
+        // set user if available (needed for mode reports)
+        userId: this.props.userName || undefined,
+        ...(!this.props.userName && { anonymousId: anonymousId }),
+        // staging or production
+        environment,
+        // pull window.location
+        location: {
+          hash: location.hash,
+          host: location.host,
+          hostname: location.hostname,
+          href: location.href,
+          origin: location.origin,
+          pathname: location.pathname,
+          search: location.search
+        }
       }
     };
-    // if user is logged in then associate feedback with them
-    // otherwise use the a random/anonymousId
-    if (this.props.userName) event.userId = this.props.userName;
-    else event.anonymousId = anonymousId;
-    // sends event to segment via forward event webhook
-    forwardEvent(event, this.props.webhook, err => {
+  }
+
+  // sends event to Segment
+  sendToSegment() {
+    // sends event to Segment via forward event webhook
+    forwardEvent(this.state.event, this.props.webhook, err => {
       if (err) {
         console.log(err); // eslint-disable-line
```

Comment:
Do you think it's worth exposing these errors so we can log them in sentry and/or provide some visual feedback to the user?

Reply:
Ohhh yes, we should!